### PR TITLE
fix: feat(WAF): WAF policy fix lint error and support new fields

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:

- Fix WAF resource policy lint error: replace deprecated function, fix format errors, modify log message and some error handling.
- WAF policy resource support new feature: 
  - add new fields `robot_action`, `bind_hosts`, `deep_inspection`, `header_inspection`, `shiro_decryption_check` and `options switch`.
  - mark field `crawler` deprecated.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- If the switch in field `options` is not specified, make them default to false.
- The API URL is : https://support.huaweicloud.com/api-waf/UpdatePolicy.html

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafPolicyV1_basic'
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafPolicyV1_basic -timeout 360m -parallel 4
=== RUN   TestAccWafPolicyV1_basic 
=== PAUSE TestAccWafPolicyV1_basic 
=== CONT  TestAccWafPolicyV1_basic 
--- PASS: TestAccWafPolicyV1_basic (368.86s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       368.923s
```

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafPolicyV1_withEpsID'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafPolicyV1_withEpsID -timeout 360m -parallel 4 
=== RUN   TestAccWafPolicyV1_withEpsID 
=== PAUSE TestAccWafPolicyV1_withEpsID
=== CONT  TestAccWafPolicyV1_withEpsID
--- PASS: TestAccWafPolicyV1_withEpsID (352.33s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       352.387s
```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go [hunk 1]
```diff
@@ -47,244 +43,346 @@ func ResourceWafPolicyV1() *schema.Resource {
...
-	d.Set("name", n.Name)
-	d.Set("level", n.Level)
-	d.Set("protection_mode", n.Action.Category)
-	d.Set("full_detection", n.FullDetection)
+	var extendRespBody interface{}
+	if extendValue, ok := n.Extend["extend"]; ok {
+		if err := json.Unmarshal([]byte(extendValue), &extendRespBody); err != nil {
+			log.Printf("[ERROR] error flatten extend map : %s", err)
+		}
+	}
```
Comment (Lance52259) [inline huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: The extendValue is never used in fields setting, so, why we need to unmarshall it?

File: docs/resources/waf_policy.md [hunk 2]
```diff
@@ -32,62 +46,106 @@ The following arguments are supported:
...
 
-* `general_check` - Indicates whether General Check in Basic Web Protection is enabled.
+* `blacklist` - (Optional, Bool) Specifies whether the Blacklist and Whitelist protection is enabled.
+  Defaults to **false**.
 
-* `crawler` - Indicates whether the master crawler detection switch in Basic Web Protection is enabled.
+* `data_masking` - (Optional, Bool) Specifies whether Data Masking is enabled. Defaults to **false**.
 
-* `crawler_engine` - Indicates whether the Search Engine switch in Basic Web Protection is enabled.
+* `false_alarm_masking` - (Optional, Bool) Specifies whether False Alarm Masking is enabled. Defaults to **false**.
```
Comment (Lance52259) [inline docs/resources/waf_policy.md]: s/Basic Web Protection/basic web protection
s/Search Engine/search engine
...

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go (hunk 1 above)
Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: The extend struct value example is: `"extend": "{\"deep_decode\":true}"`.
I add an example note for this part.

File: docs/resources/waf_policy.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/waf_policy.md]: OK

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: seems unnecessary, we can record it in the design document.

File: huaweicloud/services/waf/resource_huaweicloud_waf_policy.go [hunk 3]
```diff
@@ -47,244 +43,347 @@ func ResourceWafPolicyV1() *schema.Resource {
...
-	d.Set("region", config.GetRegion(d))
-	d.Set("name", n.Name)
-	d.Set("level", n.Level)
-	d.Set("protection_mode", n.Action.Category)
-	d.Set("full_detection", n.FullDetection)
+	// the extend struct value example is: "extend": "{\"deep_decode\":true}"
+	var extendRespBody interface{}
+	if extendValue, ok := n.Extend["extend"]; ok {
+		if err := json.Unmarshal([]byte(extendValue), &extendRespBody); err != nil {
+			log.Printf("[ERROR] error flatten extend map : %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]:
```suggestion
			log.Printf("[WARN] error flatten extend map: %s", err)
```

Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_policy.go]: OK

**Pull request merged**